### DISCUSSION TITLE
Using render.com to create a free Wiki.JS instance and database (similar to heroku)

### DISCUSSION THREAD

**Discussion by Arkanic** [requarks/wiki#6370] [Guides]
Hello all

I managed to get a wiki.js instance up and running on [render.com](https://render.com) using the free tier web app and postgresql database without much hassle. Given the recent stop of heroku free instances, I figured i'd share, as this is a perfect drop-in replacement for the heroku free wikijs instance thing.
It's pretty hacky (using the heroku docker image) but it seems to work fine once it is up and running.

how-to:

**Requirements:**
- A render.com account (no credit card, only using free tier)

### Create a postgresql instance for Wiki.JS
From the dashboard go to New > PostgreSQL
![image](https://user-images.githubusercontent.com/50847107/232985336-a1178799-468e-49ab-a357-34d398ce547e.png)
Pick a name, database name, and username with the text boxes (the specifics do not matter)
Click create database
Should take a couple of minutes to spin up the instance, once it is live scroll down to and copy "Internal Database URL" (will be used later)
![image](https://user-images.githubusercontent.com/50847107/232979008-2783e887-a093-4be3-bd36-71d1838a1390.png)

### Create Wiki.JS instance
Now go back to the dashboard and create a new "Web Service", giving it the name that you want the outward-facing url to represent (i.e. the name "fluffyunicorn" will have the provided domain "fluffyunicorn.onrender.com"
Put `https://github.com/requarks/wiki-heroku` as the repository to run, make sure the instance type is set to "Docker"
Scroll down to advanced, and create a new environment variable titled `DATABASE_URL` with the content being the link you copied earlier
![image](https://user-images.githubusercontent.com/50847107/232979515-f4a3f3a8-82d1-4f4f-a7d2-e58168898be1.png)
Press create, instance will spin up
Eventually the logs will ask you to "set up". Visit `<your project name>.onrender.com` to complete the setup as per usual. Make sure to set the hostname to your url. Once you complete close the tab and whatever proxy render is using will switch the forwarded port to the main wiki.js instance, and you now have a free instance!

## Notes
- I'll probably make a pr or something to add this to the docs, but I thought i'd see what people think about it first
- I think there is custom domain support, so if you'd prefer an EU.org subdomain or have your own that'll probably work too
- The postgresql free instance has a 1GB limit, which is likely large enough for any reasonable use case - especially given that it is zero cost
- How much  "future proofing" this setup will have given that it is the heroku-specific image is questionable, but it seems compatible enough to work due to the custom DATABASE_URL env variable being supplied. Given that heroku is no longer offering free instances, the fact that render.com works as a simple replacement is pretty useful.

**Comment by Arkanic** [comment]
update: after waiting for the instance to wind down (after half an hour or something) the system spins up again, but it is pretty slow - taking something like 30 seconds to load, where it will spit out an error about not knowing what jwt is - once you refresh again and wait another 10 seconds it fixes itself

**Comment by lalrochhara** [comment]
> Hello all
> 
> I managed to get a wiki.js instance up and running on [render.com](https://render.com) using the free tier web app and postgresql database without much hassle. Given the recent stop of heroku free instances, I figured i'd share, as this is a perfect drop-in replacement for the heroku free wikijs instance thing. It's pretty hacky (using the heroku docker image) but it seems to work fine once it is up and running.
> 
> how-to:
> 
> **Requirements:**
> 
> * A render.com account (no credit card, only using free tier)
> 
> ## Steps
> ### Create a postgresql instance for Wiki.JS
> From the dashboard go to New > PostgreSQL ![image](https://user-images.githubusercontent.com/50847107/232985336-a1178799-468e-49ab-a357-34d398ce547e.png) Pick a name, database name, and username with the text boxes (the specifics do not matter) Click create database Should take a couple of minutes to spin up the instance, once it is live scroll down to and copy "Internal Database URL" (will be used later) ![image](https://user-images.githubusercontent.com/50847107/232979008-2783e887-a093-4be3-bd36-71d1838a1390.png)
> 
> ### Create Wiki.JS instance
> Now go back to the dashboard and create a new "Web Service", giving it the name that you want the outward-facing url to represent (i.e. the name "fluffyunicorn" will have the provided domain "fluffyunicorn.onrender.com" Put `https://github.com/requarks/wiki-heroku` as the repository to run, make sure the instance type is set to "Docker" Scroll down to advanced, and create a new environment variable titled `DATABASE_URL` with the content being the link you copied earlier ![image](https://user-images.githubusercontent.com/50847107/232979515-f4a3f3a8-82d1-4f4f-a7d2-e58168898be1.png) Press create, instance will spin up Eventually the logs will ask you to "set up". Visit `<your project name>.onrender.com` to complete the setup as per usual. Make sure to set the hostname to your url. Once you complete close the tab and whatever proxy render is using will switch the forwarded port to the main wiki.js instance, and you now have a free instance!
> 
> ## Notes
> * I'll probably make a pr or something to add this to the docs, but I thought i'd see what people think about it first
> * I think there is custom domain support, so if you'd prefer an EU.org subdomain or have your own that'll probably work too
> * The postgresql free instance has a 1GB limit, which is likely large enough for any reasonable use case - especially given that it is zero cost
> * How much  "future proofing" this setup will have given that it is the heroku-specific image is questionable, but it seems compatible enough to work due to the custom DATABASE_URL env variable being supplied. Given that heroku is no longer offering free instances, the fact that render.com works as a simple replacement is pretty useful.

this is best ever tutor or guide i've ever seen this year, it's good for testing purposes and ya after decide to upgrade, good advice.

**Comment by bigman73** [comment]
The free tier is just for one month, then it gets auto deleted unless moving to a paid tier

"Your database will expire on June 27, 2025. The database will be deleted unless you [upgrade to a paid instance type.]"

**Reply by Arkanic** [reply]
must have been a recent pricing thing, was only using it as a test for a short period of time
